File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Core/RuntimeHostModules.swift
import Foundation

/// The lifecycle host surface attached to one Apple runtime host.
@MainActor
public final class LifecycleHost: LifecycleEvents {
  private let events: any LifecycleEvents

  /// Create one lifecycle host surface.
  public init(
    events: any LifecycleEvents
  ) {
    self.events = events
  }

  /// Send one lifecycle event into the attached runtime session.
  public func sendLifecycleEvent(_ event: RuntimeHostLifecycleEvent) {
    events.sendLifecycleEvent(event)
  }
}

File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/core/RuntimeHostModules.kt
package dev.destack.runtime.android.core

import dev.destack.runtime.android.module.background.BackgroundEvents
import dev.destack.runtime.android.module.background.BackgroundRequests
import dev.destack.runtime.android.module.lifecycle.LifecycleEvents
import dev.destack.runtime.android.module.text.TextEvents
import dev.destack.runtime.android.module.text.TextRequests

/**
 * The lifecycle host surface attached to one Android runtime host.
 */
public class LifecycleHost(
    events: LifecycleEvents,
) : LifecycleEvents by events
